website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
Sure. We've had this conversation before :-)  but here's the high level:

* aws-ec2 can only authorize EC2 instances and requires working with the sad parts of the EC2 API and metadata. On the flip side, because we have a lot of information about EC2 instances, we can also constrain credentials by account id, subnet id, vpc id, ami id, and all other properties. This allows you to be very granular in terms of the requirements that must be satisfied for each EC2 instance.
* aws-iam allows any AWS actor to gain credentials, which is a big pro, but on the flip side it has little granularity because other than basic caller identity you don't have much in the way of filtering.

To be clear, I see real value in both approaches. I don't have an issue with the docs indicating the benefits of the aws-iam approach over the aws-ec2 approach, but not also indicating the areas in which aws-ec2 has clear benefits is a disservice to users trying to figure out which approach works for them.